test/request-error.test.js

```diff
@@ -412,3 +415,89 @@ test('test request.routeOptions.version', t => {
     })
   })
 })
+
+test('customErrorHandler should throw for json err and stream response', t => {
```

Comment:
I don't understand, you should be able to send custom data through `reply.send`. If it cannot, it should be a bug.